clang-tools-extra/clangd/Config.h

```diff
@@ -151,6 +151,15 @@ struct Config {
     ArgumentListsPolicy ArgumentLists = ArgumentListsPolicy::FullPlaceholders;
   } Completion;
 
+  enum class HeaderInsertionPolicy {
+    IWYU,       // Include what you use
+    NeverInsert // Never insert headers as part of code completion
+  };
+
+  struct {
+    HeaderInsertionPolicy Policy = HeaderInsertionPolicy::IWYU;
+  } HeaderInsertion;
```

Comment:
Since header insertion is a side effect of accepting of code completion proposal, it would make sense to have this option under `Completion` rather than at the top level.